File: Backend/RecipesService/Models/Recipe.cs
namespace RecipesService.Models;

public class Recipe
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Title { get; set; }
    public string ShortDescription { get; set; }
    public string IngredientsList { get; set; }
    public string CookingSteps { get; set; }
    public Guid ImageId { get; set; }
    public int PrepTime { get; set; }
    public int CookTime { get; set; }
    public int Servings { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Review> Reviews{ get; set; } = new List<Review>();
}

File: frontend/src/types/Recipe.ts
type Recipe = {
    id: unknown,
    userId: any,
    title: string,
    shortDescription: string,
    ingredientsList: string, // TODO: List of strings
    cookingSteps: string, // TODO: List of strings
    imageId: string,
    prepTime: number,
    cookTime: number,
    servings: any,
    createdAt: any,
    updatedAt: any,
};

export default Recipe;